### PR TITLE
Restore backwards compatibility for some widgets

### LINKED ISSUE #426: Make some changes in AWB 2.0 backwards compatible
I've been thinking about how the rest of AiiDAlab apps will need to migrate to our new stack and AiiDA-2.0. From our experience with QeApp and my application we know that it is a bit of a pain, to say the least. Therefore I'd suggest we try to make it easier for others to migrate, otherwise we risk Python2-Python3 style split in the ecosystem.

One concrete suggestion specifically for AWB would be to make certain things backwords compatible. 
Concretely, some widgets were changed to use UUID in traitlets instead of AiiDA Nodes. That was a neccessary change, but we can easily make it backwards compatible by preserving the original traitlets, and making a small shim which converts them to UUID.

What do you think @yakutovicha @unkcpz. (LMK if my suggestion is not clear).

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #426

### CODE REVIEW

File: tests/test_process.py
Comment (unkcpz) [inline tests/test_process.py:259]:
```suggestion
    daemon_client.stop_daemon(wait=True)
```

If use `daemon_client`, need to stop daemon explicitly after the test.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @yakutovicha for taking care of this. The changes are good. 
Although there is an issue caused by using threading with daemon as we discussed offline, I propose we ignore it for the moment.

```
03/25/2023 11:26:31 AM <2172657> sqlalchemy.pool.impl.QueuePool: [ERROR] Exception during reset or similar
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 753, in _finalize_fairy
    fairy._reset(pool)
  File "/opt/conda/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 1004, in _reset
    pool._dialect.do_rollback(self)
  File "/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 683, in do_rollback
    dbapi_connection.rollback()
psycopg2.OperationalError: server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
```

I spend some time try to figure it out and narrow down it with a minimal reproduce case. It can be reproduced when submit the process using test daemon, and having a separate threading query the database. 

The test can be like this:

```python
def test_process_monitor(
    multiply_add_process_builder_ready, daemon_client, await_for_process_completeness
):
    """Test ProcessMonitor with a simple `WorkChainNode`"""
    from aiidalab_widgets_base.process import ProcessMonitor

    process = engine.submit(multiply_add_process_builder_ready)
    widget = ProcessMonitor(value=process.uuid)
```

The `_monitor_process` method of `ProcessMonitor` just need to have a query as:

```python
    def _monitor_process(self, process_uuid):
        assert process_uuid is not None
        process = orm.load_node(process_uuid)
```

I assume this still comes from aiida in general not thread-safe, I think maybe we need help from Chris to dig the root out. I'll ask him and set a meeting for that, but this has to be after the 2.0 release of AWB.

**Comment by danielhollas** [comment]
@yakutovicha I will take a closer look at this, please wait before merging, thanks.

**Comment by danielhollas** [comment]
>  Although there is an issue caused by using threading with daemon as we discussed offline, I propose we ignore it for the moment.

@unkcpz can you elaborate on this? Is this problem present also on master or only on this PR? If the latter, we should not merge this.
What aiida-core version have you seen it? I think there have been some issues with the daemon for version < 2.2.2

**Review by danielhollas — COMMENTED** [review]
tbh I am not very happy about the code here (and I realize I was the one pushing for this change, my apologies). The circular nature of the `value` and `process` traits makes the code very confusing and error prone imo. My original idea was to have a one-way flow from `process` to `value`, but that is not great either since they would not be in sync which might lead to bugs as well.

**Comment by danielhollas** [comment]
@yakutovicha that is fine with me, since you anyway said we have currently control over all apps in the appstore.

Btw: the new test in this PR might be worthwhile merging.

I am travelling today back from Switzerland, should be available more tomorrow.

**Comment by unkcpz** [comment]
> @unkcpz can you elaborate on this? Is this problem present also on master or only on this PR? If the latter, we should not merge this.
What aiida-core version have you seen it? I think there have been some issues with the daemon for version < 2.2.2

The error is shown in  https://github.com/aiidalab/aiidalab-widgets-base/pull/471#pullrequestreview-1357821001 and can be reproduce with the latest docker stack which running with aiida-core 2.2.2. 

It is not this PR which introduces the issue, it was there already. The tests will not failed but the error messages popped up showing the DB session is not properly shut down after the query. @yakutovicha and I suspect this might be the issue that causes the memory leak of the container we run for the moment.

**Comment by unkcpz** [comment]
Seems we don't want to continue on this route. 
So @yakutovicha can you update the PR to leave only the test as @danielhollas suggested?

**Comment by yakutovicha** [comment]
> Seems we don't want to continue on this route. So @yakutovicha can you update the PR to leave only the test as @danielhollas suggested?

yep, I just opened PR #478 with the test only.

**Pull request closed without merging**
